Stormshield/stormshield_module/base.py

```diff
@@ -0,0 +1,97 @@
+from functools import cached_property
+from posixpath import join as urljoin
+from typing import Any
+import re
+import requests
+from requests import RequestException, Response
+from tenacity import (
+    RetryError,
+    Retrying,
+    retry_if_exception_type,
+    stop_after_attempt,
+)
+
+from sekoia_automation.action import GenericAPIAction
+
+
+class StormshieldAction(GenericAPIAction):
+    endpoint: str
+
+    @cached_property
+    def api_token(self):  # type: ignore
+        return self.module.configuration.get("api_token")
+
+    @cached_property
+    def base_url(self) -> str:
+        config_url = self.module.configuration["url"].rstrip("/")
+        api_path = "rest/api/v1"
+        return urljoin(config_url, api_path)
+
+    def get_headers(self) -> dict[str, str]:
+        return {"Authorization": f"Bearer {self.api_token}"}
+
+    def treat_failed_response(self, response: Response) -> None:
+        errors = {
+            401: "Authentication failed: Invalid API key provided.",
+            403: "Access denied: Insufficient permissions to access this resource.",
+            404: "Resource not found: The requested resource could not be located.",
+            409: "Conflict detected: The specified agent is not compatible with this feature.",
+            500: "Internal server error: Rate limit exceeded.",
+        }
+
+        message = errors.get(response.status_code)
+
+        if message:
+            raise Exception(f"Error : {message}")
+
+    def get_url(self, arguments: dict[str, Any]) -> str:
+        match = re.findall("{(.*?)}", self.endpoint)
+        for replacement in match:
+            self.endpoint = self.endpoint.replace(f"{{{replacement}}}", str(arguments.pop(replacement)), 1)
+
+        path = urljoin(self.base_url, self.endpoint.lstrip("/"))
+
+        if self.query_parameters:
+            query_arguments: list[str] = []
+
+            for k in self.query_parameters:
+                if k in arguments:
+                    value = arguments.pop(k)
+                    if isinstance(value, bool):
+                        value = int(value)
```

Comment:
For boolean, the StormShield API only accept `true` or `false`. Convert the boolean into string.
```suggestion
                        value = str(value)
```